gcomm/src/view.cpp

```diff
@@ -247,12 +247,25 @@ std::istream& gcomm::ViewState::read_stream(std::istream& is)
     return is;
 }
 
-void gcomm::ViewState::write_file(const char* fname) const
+std::string gcomm::ViewState::get_viewstate_file_name(gu::Config& conf)
+{
+    std::string dir_name = COMMON_BASE_DIR_DEFAULT;
+
+    try {
+        // If base_dir is set in the configuration we should use
+        // it instead of current directory default.
+        dir_name = conf.get(COMMON_BASE_DIR_KEY, dir_name);
+    } catch (const gu::NotFound &) {
+        // In case it is not known we do not have to do
+        // anything and use default.
```

Comment:
The absence of the option in the configuration object is a sign of a serious error, because default value should always be there. So I would print an error message and rethrow in optimised code and assert() in debug. Or perhaps the whole try() block is unnecessary.